Makefile

```diff
@@ -4,12 +4,15 @@ regress_dir ?= $(root_dir)/regress
 firrtl_dir ?= $(root_dir)/src/main/stanza
 install_dir ?= $(root_dir)/utils/bin
 
+sbt ?= sbt
 stanza ?= $(install_dir)/stanza
 stanza_bin ?= $(install_dir)/firrtl-stanza
 scala_jar ?= $(install_dir)/firrtl.jar
-scala_src=$(shell ls src/main/scala/firrtl/*.scala)
+# The following is slow but accurate.
+# We use 'unmanaged' so we don't include source files generated by the
+#  build itself (otherwise, we'd need to include the rules to build them).
+scala_src := $(shell $(sbt) "show compile:unmanaged-sources compile:unmanaged-resources" | ./sourcesFromSbt.sh )
```

Comment:
If you've built once,

    find . -type f -name "*.scala"

will pick up the managed (generated sources) and it won't pick up resources.

    find src -type f \( -name "*.scala" -o -path "*/resources/*" \)

should pick up almost everything.

I'd be willing to sacrifice accuracy for speed in this case. You can always force a complete build by doing a

    make clean build

> On Apr 7, 2016, at 1:19 PM, Andrew Waterman <notifications@github.com> wrote:
> 
> In Makefile:
> 
> >  stanza ?= $(install_dir)/stanza
> >  stanza_bin ?= $(install_dir)/firrtl-stanza
> >  scala_jar ?= $(install_dir)/firrtl.jar
> > -scala_src=$(shell ls src/main/scala/firrtl/*.scala)
> > +# The following is slow but accurate.
> > +# We use 'unmanaged' so we don't include source files generated by the
> > +#  build itself (otherwise, we'd need to include the rules to build them).
> > +scala_src := $(shell $(sbt) "show compile:unmanaged-sources compile:unmanaged-resources" | ./sourcesFromSbt.sh )
> 
> I'm glad we're fixing this. Can we just do
> 
> find . -type f -name "*.scala".
> 
> to avoid invoking sbt twice?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>